src/main/java/maksym/fedorenko/bookstore/repository/BookRepository.java

```diff
@@ -1,10 +1,26 @@
 package maksym.fedorenko.bookstore.repository;
 
+import com.querydsl.core.types.Predicate;
+import java.util.List;
+import java.util.Optional;
 import maksym.fedorenko.bookstore.model.Book;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.EntityGraph;
 import org.springframework.data.jpa.repository.JpaRepository;
 import org.springframework.data.querydsl.QuerydslPredicateExecutor;
 import org.springframework.stereotype.Repository;
 
 @Repository
 public interface BookRepository extends JpaRepository<Book, Long>, QuerydslPredicateExecutor<Book> {
+    @EntityGraph(attributePaths = "categories")
+    Optional<Book> findById(Long id);
+
+    @EntityGraph(attributePaths = "categories")
+    Page<Book> findAll(Pageable pageable);
+
+    @EntityGraph(attributePaths = "categories")
+    Page<Book> findAll(Predicate predicate, Pageable pageable);
```

Comment:
just curious

do you really need to load categories eagerly in each of this situations?

if yes - feel free to skip this comment

Reply:
I just have two places where I need to load categories with a book and then map them into a list of category IDs. This is mainly needed in book endpoints. But in the "/categories/{id}/books" requirements are written that I don't need to have information about categories. So I decided to create custom methods with loading categories.